lib/components/form/batch-settings.tsx

```diff
@@ -223,6 +225,10 @@ function BatchSettings({
   )
   const accentColor = tinycolor(baseColor).darken(10)
 
+  const _onSettingsUpdate = (params: any) => {
+    setQueryParam({ queryParamData: params, ...params })
```

Comment:
Should `setQueryParam` be used for callback `_toggleModeButton` too?

Reply:
We could use it for that! I don't know of any sync bugs affecting that, I think because those keys are not in conflict with the ones defined in query-params.js in core-utils. But I think it wouldn't hurt to use it and would probably make things more resilient. I'll try it out.